guvnor-ala/guvnor-ala-registry-local/src/main/java/org/guvnor/ala/registry/local/InMemoryBuildRegistry.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2016 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.guvnor.ala.registry.local;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.enterprise.context.ApplicationScoped;
+
+import org.guvnor.ala.build.Binary;
+import org.guvnor.ala.registry.BuildRegistry;
+
+/**
+ * @TODO: This is a not thread-safe implementation for local testing. A
+ * more robust and distributed implementation should be provided for real
+ * use cases. All the lookups mechanisms and structures needs to be improved for performance.
+ */
```

Comment:
The map seems to be the only mutable state of this class, so we could use a concurrent hash map instead re: thread-safety comment?